tool/src/java/org/sakaiproject/gradebookng/tool/model/GradebookUiSettings.java

```diff
@@ -49,7 +50,8 @@
 	public GradebookUiSettings() {
 		//defaults. Note there is no default for assignmentSortOrder as that requires an assignmentId which will differ between gradebooks
 		this.categoriesEnabled = false;
-		this.assignmentVisibility = new HashMap<Long, Boolean>();
+		this.assignmentVisibility = new ConcurrentHashMap<Long, Boolean>();
+		this.categoryScoreVisiblity = new ConcurrentHashMap<String, Boolean>();
 		this.nameSortOrder = GbStudentNameSortOrder.LAST_NAME;
```

Comment:
The use of ConcurrentHashMap immediately raised red flags for me. Clicking 'hide all' just toggles all checkboxes to be clicked and hence fires their onUpdate method correct? I haven't applied this yet, what do you see, seems ok from here.

Perhaps hideAll/showAll could handle it a bit differently and not call each in sequence but rather just one method to update the map all at once?